### PR TITLE
[SM68] Add AST test for [Empty]NodeOutputArray types

### PR DESCRIPTION
**Pull request by python3kgae**
Fixes https://github.com/microsoft/DirectXShaderCompiler/issues/6199

### CODE REVIEW

File: tools/clang/test/SemaHLSL/hlsl/workgraph/ast-EmptyNodeOutputArrayTypes.hlsl [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// CHECK-NEXT:| |-FinalAttr 0x{{.+}} <<invalid sloc>> Implicit final
+// CHECK-NEXT:| |-HLSLNodeObjectAttr 0x{{.+}} <<invalid sloc>> Implicit EmptyNodeOutputArray
+// CHECK-NEXT:| |-FieldDecl 0x{{.+}} <<invalid sloc>> <invalid sloc> implicit h 'int'
+// CHECK-NEXT:| `-CXXMethodDecl 0x[[SUB:[0-9a-f]+]] <<invalid sloc>> <invalid sloc> used operator[] 'EmptyNodeOutput (unsigned int)'
+// CHECK-NEXT:|   |-ParmVarDecl 0x{{.+}} <<invalid sloc>> <invalid sloc> index 'unsigned int'
+// CHECK-NEXT:|   |-HLSLIntrinsicAttr 0x{{.+}} <<invalid sloc>> Implicit "indexnodehandle" "" 12
+// CHECK-NEXT:|   `-HLSLCXXOverloadAttr 0x{{.+}} <<invalid sloc>> Implicit
+
+// CHECK: `-FunctionDecl 0x{{.+}} node_2_0 'void (EmptyNodeOutputArray)'
+// CHECK:   |-ParmVarDecl 0x[[Param:[0-9a-f]+]] {{.+}} used OutputArray_2_0 'EmptyNodeOutputArray'
```
Comment (tex3d) [inline tools/clang/test/SemaHLSL/hlsl/workgraph/ast-EmptyNodeOutputArrayTypes.hlsl:56]: You stopped using `CHECK-NEXT` for the rest of these.  That's different than the way the use was checked in the other test.  Was that on purpose?

**Review by tex3d — APPROVED** [review]
LGTM.

**Review by pow2clk — APPROVED** [review]
Thanks for taking this on!
